mantle/platform/qemu.go

```diff
@@ -136,7 +134,9 @@ func (inst *QemuInstance) Destroy() {
 	if inst.tmpConfig != "" {
 		os.Remove(inst.tmpConfig)
 	}
-	if inst.qemu != nil {
+	// check if qemu is dead before trying to kill it
+	_, notFound := os.FindProcess(int(inst.Pid()))
```

Comment:
But wait, we're the only thing that should be calling `waitpid()` on it, so it can't have gone away - it should be safe to call `os.Kill()` even on a zombie right?

(really what we want are [pidfds](https://lwn.net/Articles/801319/) but we can't rely on those yet)

Reply:
The reason for checking that was otherwise there's an exit error for `Wait()` having already been called.

Reply:
Oh right of course, because the main process blocks in `Wait()`.  So I think the right way to fix this though is to nilify the pointers after calling `Wait()`.  We already guard against `nil` above:

```
   if inst.qemu != nil {
       inst.qemu.Wait()
       inst.qemu = nil
   }
```

And the same for the `inst.nbdServers`, clear out the array via `inst.nbdServers = nil` afterwards.

Reply:
In Rust¹, this would be `Option<Process>` and we'd do something like:

```
if let Some(proc) = inst.qemu.take() {
   proc.wait()
}
```

And the "move semantics" here is a lot clearer.

Reply:
Updated that. `exec.cmd.Kill()` already calls `Wait()` anyway. https://github.com/coreos/mantle/blob/cl/system/exec/exec.go#L71